src/tests/api/auth/sign-in/validate.spec.ts

```
@@ -0,0 +1,285 @@
+import { validate } from "api/auth/sign-in/validate";
+import { CustomError } from "utils/error";
+import { StatusCodeEnum } from "enums/status-code";
+
+describe("auth > sign-in > service", () => {
+	const validEmail = "johndoe@foo.foo";
+	const validPassword = "JohnDoe@123";
+
+	describe("Success", () => {
+		it("should return validated params", async () => {
+			let result: any;
+
+			try {
+				result = await validate({ email: validEmail, password: validPassword });
+			} catch (err) {
+				result = err;
+			}
+
+			expect(result).toStrictEqual({
+				email: validEmail,
+				password: validPassword,
+			});
+		});
+	});
+
+	describe("Invalid Params", () => {
+		it("should throw error with invalid password have only lowercase", async () => {
+			let result: any;
+
+			try {
+				result = await validate({
+					email: validEmail,
+					password: "johndoe@123",
+				});
+			} catch (err: any) {
+				result = err;
+			}
+
+			expect(result).toBeInstanceOf(CustomError);
+			expect(result.statusCode).toBe(StatusCodeEnum.BAD_REQUEST);
+			expect(result.message).toBe(
+				'password must match the following: "/^(?=.*[a-z])(?=.*[A-Z])(?=.*\\d)(?=.*[@#$!%*?&])[A-Za-z\\d@$!%*#?&]{6,100}$/"',
+			);
+		});
+
+		it("should throw error with invalid password (only uppercase)", async () => {
```

`it("should throw an error with invalid password (whitout lowercase and special characters)"`